### DISCUSSION TITLE
Enable Link-Time Optimization (LTO)

### DISCUSSION THREAD

**Discussion by zamazan4ik** [tokio-rs/toasty#33] [Ideas]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. If we enable it on the Cargo profile level, users, who install the application with `cargo install`, will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48) or more close example for the Tokio ecosystem - the `console` utility profile: https://github.com/tokio-rs/console/blob/main/Cargo.toml#L32 

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

I have made quick tests (Fedora 40) by adding `lto = true` to the Release profile. The binary size reduction is the following:
* `toasty-cli`: from 4.5 Mib to 3.9 Mib

Thank you.

**Comment by carllerche** [comment]
That makes sense. Could you open a PR for this?

**Reply by zamazan4ik** [reply]
Which way do you prefer: from `cargo-outdated` or from Tokio's `console`?

**Reply by zamazan4ik** [reply]
@carllerche friendly reminder :)
